File: bindings/java/examples/Quickstart.java
import io.vietasr.Pipeline;
import io.vietasr.Result;

public final class Quickstart {
    public static void main(String[] args) {
        if (args.length < 1) {
            System.err.println("usage: Quickstart <wav-file>");
            System.err.println("presets: " + String.join(", ", Pipeline.listPresets()));
            System.err.println("modules: " + String.join(", ", Pipeline.listModules()));
            System.exit(1);
        }
        try (Pipeline pipe = Pipeline.preset("transcribe")) {
            Result result = pipe.transcribe(args[0]);
            System.out.println(result.text());
        }
    }
}

File: bindings/rust/examples/quickstart.rs
use std::process::exit;

fn main() {
    let args: Vec<String> = std::env::args().collect();
    if args.len() < 2 {
        eprintln!("usage: quickstart <wav-file>");
        eprintln!("presets: {:?}", vietasr::list_presets());
        eprintln!("modules: {:?}", vietasr::list_modules());
        exit(1);
    }

    let pipeline = match vietasr::Pipeline::preset("transcribe") {
        Ok(p) => p,
        Err(e) => {
            eprintln!("preset: {e}");
            exit(2);
        }
    };

    match pipeline.transcribe_file(&args[1]) {
        Ok(result) => println!("{}", result.text()),
        Err(e) => {
            eprintln!("transcribe: {e}");
            exit(3);
        }
    }
}
